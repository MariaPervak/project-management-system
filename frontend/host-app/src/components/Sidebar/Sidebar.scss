.sidebar{
  padding: 25px;
  width: 200px;
  height: calc(100vh - 91px);
}