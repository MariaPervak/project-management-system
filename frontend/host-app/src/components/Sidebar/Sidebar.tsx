import { Link } from "react-router-dom";
import './Sidebar.scss';


const Sidebar = () => {
  return (
    <div>
      <nav className='sidebar'>
        <ul>
          <li><Link to="/">Список задач</Link></li>
          <li><Link to="/kanban">Kanban</Link></li>
          <li><Link to="/create">Создать</Link></li>
        </ul>
      </nav>
    </div>
  );
};

export default Sidebar;