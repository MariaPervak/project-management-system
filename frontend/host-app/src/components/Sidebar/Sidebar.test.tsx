import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import Sidebar from './Sidebar';
import { AuthContext } from '../../../context/AuthContext/AuthContext';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    Link: ({ to, children }: { to: string; children: React.ReactNode }) => (
        <a href={to}>{children}</a>
    ),
}));

describe('Sidebar component', () => {
    const renderWithRouterAndContext = (role: string = 'user') => {
        const authValue = { role };
        return render(
            <Router>
                <AuthContext.Provider value={authValue}>
                    <Sidebar />
                </AuthContext.Provider>
            </Router>
        );
    };

    it('renders basic links for all users', () => {
        renderWithRouterAndContext();

        expect(screen.getByText('Список задач')).toBeInTheDocument();
        expect(screen.getByText('Список задач')).toHaveAttribute('href', '/');
    });

    it('renders admin link when user is admin', () => {
        renderWithRouterAndContext('admin');

        expect(screen.getByText('Создать')).toBeInTheDocument();
        expect(screen.getByText('Создать')).toHaveAttribute('href', '/create');
    });

    it('does not render admin link for regular users', () => {
        renderWithRouterAndContext('user');

        expect(screen.queryByText('Создать')).not.toBeInTheDocument();
    });

    it('has correct sidebar class', () => {
        renderWithRouterAndContext();

        expect(screen.getByRole('navigation')).toHaveClass('sidebar');
    });

    it('matches snapshot for regular user', () => {
        const { asFragment } = renderWithRouterAndContext('user');
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot for admin', () => {
        const { asFragment } = renderWithRouterAndContext('admin');
        expect(asFragment()).toMatchSnapshot();
    });
});