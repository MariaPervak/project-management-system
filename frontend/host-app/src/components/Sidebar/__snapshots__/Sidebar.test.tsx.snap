// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component matches snapshot for admin 1`] = `
<DocumentFragment>
  <nav
    class="sidebar"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          Список задач
        </a>
      </li>
      <li>
        <a
          href="/create"
        >
          Создать
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Sidebar component matches snapshot for regular user 1`] = `
<DocumentFragment>
  <nav
    class="sidebar"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          Список задач
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
